sklearn/decomposition/_kernel_pca.py

```diff
@@ -434,6 +432,9 @@ def fit(self, X, y=None):
         K = self._get_kernel(X)
         self._fit_transform(K)
 
+        if self.fit_inverse_transform and self.kernel == "precomputed":
+            raise ValueError("Cannot fit_inverse_transform with a precomputed kernel.")
```

Comment:
Please move this validation to the beginning of the `fit` method, so that it's done before `_validate_data`.

Reply:
Please also make sure this is tested in the test file.